src/input/type_repository.hh

```diff
@@ -77,7 +77,17 @@ public:
      * finished and then in second step can be finished
      * other types.
      */
-    void finish(bool is_root_of_generic_subtree = false);
+    void finish(Type::FinishType finish_type = Type::FinishType::regular);
```

Comment:
Add to doc. That this is meant to be used only with FinishType::delete.
Remove default value for finish_type.